docs/utilities/large_messages.md

```diff
@@ -0,0 +1,322 @@
+specifying the size of the offloaded message and the message contains a pointer to the S3 object.
+
+If this is the case, the utility will retrieve the object from S3 using the `getObject(bucket, key)` API,
+and place the content of the object in the message payload. You can then directly use the content of the message.
+If there was an error during the S3 download, the function will fail with a `LargeMessageProcessingException`.
+
+After your code is invoked and returns without error, the object is deleted from S3
+using the `deleteObject(bucket, key)` API. You can disable the deletion of S3 objects with the following configuration:
+
+=== "Don't delete S3 Objects"
+```java
+@LargeMessage(deleteS3Object = false)
+private void processRawMessage(SQSEvent.SQSMessage sqsMessage) {
+// do something with the message
+}
+```
+
+!!! tip
+This utility works perfectly together with the batch module (`powertools-batch`), especially for SQS:
+
+    ```java hl_lines="2 5-7 12 15 16" title="Combining batch and large message modules"
+    public class SqsBatchHandler implements RequestHandler<SQSEvent, SQSBatchResponse> {
+        private final BatchMessageHandler<SQSEvent, SQSBatchResponse> handler;
+    
+        public SqsBatchHandler() {
+            handler = new BatchMessageHandlerBuilder()
+                    .withSqsBatchHandler()
+                    .buildWithRawMessageHandler(this::processMessage);
+        }
+
+        @Override
+        public SQSBatchResponse handleRequest(SQSEvent sqsEvent, Context context) {
+            return handler.processBatch(sqsEvent, context);
+        }
+
+        @LargeMessage
+        private void processMessage(SQSEvent.SQSMessage sqsMessage) {
+            // do something with the message
+        }
+    }
+    ```
+
+## Customizing S3 client configuration
+
+To interact with S3, the utility creates a default S3 Client :
+
+=== "Default S3 Client"
+```java
+S3Client client = S3Client.builder()
+.httpClient(UrlConnectionHttpClient.builder().build())
+.region(Region.of(System.getenv(AWS_REGION_ENV)))
+.build();
+```
+
+If you need to customize this `S3Client`, you can leverage the `LargeMessageConfig` singleton:
+
+=== "Custom S3 Client"
+```java hl_lines="6"
+import software.amazon.lambda.powertools.largemessages.LargeMessage;
+
+    public class SnsRecordHandler implements RequestHandler<SNSEvent, String> {
+        
+        public SnsRecordHandler() {
+            LargeMessageConfig.init().withS3Client(/* put your custom S3Client here */);
+        }
+
+        @Override
+        public String handleRequest(SNSEvent event, Context context) {
+            processSNSRecord(event.records.get(0)); 
+            return "Hello World";
+        }
+
+        @LargeMessage
+        private void processSNSRecord(SNSEvent.SNSRecord snsRecord) {
+            // snsRecord.getSNS().getMessage() will contain the content of the S3 object
+        }
+    }
+    ```
+
+## Migration from the SQS Large Message utility
+
```

Comment:
Maybe add something saying "If you are still using the powertools-sqs library for batch processing, consider moving to `powertools-batch` at the same time to remove the dependency on this library completely; it has been deprecated will be removed in v2"